tests/extension/oneapi_device_global/device_global_functional_spec_constants.cpp

```diff
@@ -0,0 +1,181 @@
+    // Default value of type T in case if we expect to read default value
+    T def_val{};
+    // Changed value of type T in case if we expect to read modified value
+    T new_val{};
+    value_helper<T>::change_val(new_val);
+
+    constexpr int initial_sc_val = 1;
+    constexpr int changed_sc_val = 2;
+    const int sc_val = is_def_val_expected ? initial_sc_val : changed_sc_val;
+
+    // is_read_correct will be set to true if device_global value is equal to
+    // the expected_val inside kernel
+    bool is_read_correct{false};
+
+    {
+      // Creating result buffer
+      sycl::buffer<bool, 1> is_read_corr_buf(&is_read_correct,
+                                             sycl::range<1>(1));
+      auto queue = util::get_cts_object::queue();
+      queue.submit([&](sycl::handler& cgh) {
+        using kernel = kernel_read_then_write<T>;
+
+        cgh.template set_specialization_constant<spec_const_id>(sc_val);
+
+        auto is_read_correct_acc =
+            is_read_corr_buf.template get_access<sycl::access_mode::write>(cgh);
+
+        // Kernel that will compare device_global variable with expected and
+        // then write new value from specialization constants
+        cgh.single_task<kernel>([=](sycl::kernel_handler h) {
+          if (is_def_val_expected) {
+            is_read_correct_acc[0] =
+                value_helper<T>::compare_val(dev_global<T>, def_val);
+          } else {
+            is_read_correct_acc[0] =
+                value_helper<T>::compare_val(dev_global<T>, new_val);
+          }
+
+          // Get specialization constant value for change device_global instance
+          int sc_val = h.template get_specialization_constant<spec_const_id>();
+          value_helper<T>::change_val(dev_global<T>, sc_val);
+        });
+      });
+      queue.wait_and_throw();
+    }
+    if (is_read_correct == false) {
+      FAIL(log, get_case_description(
+                    "device_global: Interaction with specialization constants",
+                    error_info, type_name));
+    }
+  }
+};
+
+/**
+ * @brief The function tests that the device_global value is correctly read and
+ * changed while interacting with specialization constants
+ * @tparam T Type of underlying device_global value
+ */
+template <typename T>
+void run_test(util::logger& log, const std::string& type_name) {
+  using VerifierT = read_and_write_in_kernel<T>;
+  // At the first run expect default value
+  VerifierT::expect_def_val(log, type_name);
+
+  // At the second run expect changed value
+  VerifierT::expect_new_val(log, type_name);
+}
+}  // namespace spec_constants_interaction
+
+template <typename T>
+class check_device_global_spec_constants {
+ public:
+  void operator()(sycl_cts::util::logger& log, const std::string& type_name) {
+    spec_constants_interaction::run_test<T>(log, type_name);
+    spec_constants_interaction::run_test<T[5]>(log, type_name);
+  }
+};
+#endif
+
+/** test device_global functional
```

Comment:
What do you mean by `functional`?

Reply:
By `functional`, I mean logic that the `device_global` should provide. 
As I see it, `device_global` will provide device scoped memory allocations that can be accessed within a kernel using syntax similar to C global variables. 
So, by `functional` tests, I want to confirm, that such a mechanism works. 
In this concrete case, I test that `device_global` interacts with specialization constants properly.

Reply:
OK, I understand the meaning, but as I am not English native I pass. :-)